verifier-sdk/src/light_transaction.rs

```diff
@@ -156,12 +154,20 @@ impl<
     /// verifies the zero knowledge proof, inserts leaves, inserts nullifiers, transfers funds and fees.
     #[inline(never)]
     pub fn transact(&mut self) -> Result<()> {
+        #[cfg(all(target_os = "solana", feature = "mem-profiling"))]
+        custom_heap::log_total_heap("pre load MerkleTreeSet");
+        // let mut merkle_tree_set = self.input.ctx.accounts.get_merkle_tree_set().load_mut()?;
+        #[cfg(all(target_os = "solana", feature = "mem-profiling"))]
+        custom_heap::log_total_heap("post load MerkleTreeSet");
+
+        // self.emit_indexer_transaction_event(&mut merkle_tree_set)?;
         self.emit_indexer_transaction_event()?;
         #[cfg(all(target_os = "solana", feature = "mem-profiling"))]
         custom_heap::log_total_heap("emit_indexer_transaction_event");
         self.compute_tx_integrity_hash()?;
         #[cfg(all(target_os = "solana", feature = "mem-profiling"))]
         custom_heap::log_total_heap("compute_tx_integrity_hash");
+        // self.fetch_root(&mut merkle_tree_set)?;
```

Comment:
leftover comment

Reply:
Whoops, sorry.

Fixed. And I also added heap logs in the new places where I'm loading the MTS.

For more context - to get rid of "account already borrowed" error, I had to make sure that `merkle_tree_set` variables are dropped before doing CPI. Reading from MTS and then doing a CPI (with MTS as one of accounts) without dropping the MTS variable triggers the issue (which makes sense).